src/components/molecules/BenefitFilter.test.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+
+import {
+  Primary,
+  Eligible,
+  PotentialHelp,
+  Others,
+} from "./BenefitFilter.stories";
+
+it("renders BenefitFilter in its primary state", () => {
+  render(<Primary {...Primary.args} />);
+  const text = screen.getByText("Filter by");
+  const eligible = screen.getByText("Eligible benefits");
+  const help = screen.getByText("Potential help");
+  const others = screen.getByText("Others");
+  const eligibleCount = screen.getByText("15");
+  const helpCount = screen.getByText("8");
+  const othersCount = screen.getByText("30");
+  expect(text).toBeTruthy();
+  expect(eligible).toBeTruthy();
+  expect(help).toBeTruthy();
+  expect(others).toBeTruthy();
+  expect(eligibleCount).toBeTruthy();
+  expect(helpCount).toBeTruthy();
+  expect(othersCount).toBeTruthy();
+  expect(Primary.args.isSelectedEligible).toBe(false);
```

Comment:
Useless test here, again you're not testing what is rendered with this expect. What you are doing is testing that you set `Primary.args.isSelectedEligible` to false in the story. Since you you were able to get the eligible button you should be able to test whether it's selected or not by the presence of certain css classes

Reply:
Refactored the tests, I used getByAttribute to check if the id matches, also tested the buttons in its selected stage by checking if certain css contains in the class list